src/slurm_plugin/clustermgtd.py

```diff
@@ -596,36 +713,38 @@ def _terminate_orphaned_instances(self, cluster_instances, ips_used_by_slurm):
         instances_to_terminate = []
         for instance in cluster_instances:
             if instance.private_ip not in ips_used_by_slurm and time_is_up(
-                instance.launch_time, self.current_time, self.sync_config.orphaned_instance_timeout
+                instance.launch_time, self._current_time, self._config.orphaned_instance_timeout
             ):
                 instances_to_terminate.append(instance.id)
         if instances_to_terminate:
             log.info("Terminating the following orphaned instances: %s", instances_to_terminate)
-            self.instance_manager.delete_instances(
-                instances_to_terminate, terminate_batch_size=self.sync_config.terminate_max_batch_size
+            self._instance_manager.delete_instances(
+                instances_to_terminate, terminate_batch_size=self._config.terminate_max_batch_size
             )
 
 
 def _run_clustermgtd():
     """Run clustermgtd actions."""
-    cluster_manager = ClusterManager()
+    config = ClustermgtdConfig(os.path.join(CONFIG_FILE_DIR, "parallelcluster_clustermgtd.conf"))
```

Comment:
Let's remove the `__repl__` call in `ClustemgtdConfig.__init__`, and have manage_cluster log the config so we don't have 2 logs of config at the start?

Reply:
done! I also slightly optimized this logic so that it checks if the new config is different from the previous one before re-initializing the objects and writing a log message.